### ISSUE TITLE
[Event] codeunit 12 "Gen. Jnl.-Post Line" OnAfterVendLedgEntryInsert

### ISSUE THREAD

**Issue by pri-kise** [microsoft/ALAppExtensions#26303]
### Describe the request

Why did you obsolete the following event in BC23?
```al
#if not CLEAN23
    [Obsolete('This event is obsolete. Use OnAfterVendLedgEntryInsertInclPreviewMode instead.', '23.0')]
    [IntegrationEvent(true, false)]
    local procedure OnAfterVendLedgEntryInsert(var VendorLedgerEntry: Record "Vendor Ledger Entry"; GenJournalLine: Record "Gen. Journal Line"; var DtldLedgEntryInserted: Boolean)
    begin
    end;
#endif
    [IntegrationEvent(true, false)]
    local procedure OnAfterVendLedgEntryInsertInclPreviewMode(var VendorLedgerEntry: Record "Vendor Ledger Entry"; GenJournalLine: Record "Gen. Journal Line"; var DtldLedgEntryInserted: Boolean; PreviewMode: Boolean)
    begin
    end;
```

Could you please instead add the parameter `PreviewMode: Boolean` to the existing event `OnAfterVendLedgEntryInsert` and revert the obsoletion?

```al
    [IntegrationEvent(true, false)]
    //+Change
    //-local procedure OnAfterVendLedgEntryInsert(var VendorLedgerEntry: Record "Vendor Ledger Entry"; GenJournalLine: Record "Gen. Journal Line"; var DtldLedgEntryInserted: Boolean)
    local procedure OnAfterVendLedgEntryInsert(var VendorLedgerEntry: Record "Vendor Ledger Entry"; GenJournalLine: Record "Gen. Journal Line"; var DtldLedgEntryInserted: Boolean; PreviewMode: Boolean)
    //-Change
    begin
    end;
    [IntegrationEvent(true, false)]
    local procedure OnAfterVendLedgEntryInsertInclPreviewMode(var VendorLedgerEntry: Record "Vendor Ledger Entry"; GenJournalLine: Record "Gen. Journal Line"; var DtldLedgEntryInserted: Boolean; PreviewMode: Boolean)
    begin
    end;
```

### Additional context

It's no breaking change to add a parameter to an existing event?
This only results in much code duplication, since any partner must change the event subscriber.

```al
    local procedure PostVend(GenJournalLine: Record "Gen. Journal Line"; Balancing: Boolean)
    var
        Vend: Record Vendor;
        VendPostingGr: Record "Vendor Posting Group";
        VendLedgEntry: Record "Vendor Ledger Entry";
        CVLedgEntryBuf: Record "CV Ledger Entry Buffer";
        TempDtldCVLedgEntryBuf: Record "Detailed CV Ledg. Entry Buffer" temporary;
        DtldVendLedgEntry: Record "Detailed Vendor Ledg. Entry";
        PurchSetup: Record "Purchases & Payables Setup";
        PayablesAccount: Code[20];
        DtldLedgEntryInserted: Boolean;
        CheckExtDocNoHandled: Boolean;
        IsHandled: Boolean;
    begin
       ...

        if PurchSetup."Copy Vendor Name to Entries" then
            VendLedgEntry."Vendor Name" := Vend.Name;
        OnBeforeVendLedgEntryInsert(VendLedgEntry, GenJournalLine, GLReg);
        VendLedgEntry.Insert(true);

        VendLedgEntry.CopyLinks(GenJournalLine);

        // Post detailed vendor entries
        OnPostVendOnBeforePostDtldVendLedgEntries(VendLedgEntry, GenJournalLine, TempDtldCVLedgEntryBuf, NextEntryNo);
        DtldLedgEntryInserted := PostDtldVendLedgEntries(GenJournalLine, TempDtldCVLedgEntryBuf, VendPostingGr, true);

#if not CLEAN23
        OnAfterVendLedgEntryInsert(VendLedgEntry, GenJournalLine, DtldLedgEntryInserted);
#endif        
        OnAfterVendLedgEntryInsertInclPreviewMode(VendLedgEntry, GenJournalLine, DtldLedgEntryInserted, PreviewMode);

        if DtldLedgEntryInserted then
            if IsTempGLEntryBufEmpty() then
                DtldVendLedgEntry.SetZeroTransNo(NextTransactionNo);
        DeferralPosting(GenJournalLine."Deferral Code", GenJournalLine."Source Code", PayablesAccount, GenJournalLine, Balancing);

        OnMoveGenJournalLine(GenJournalLine, VendLedgEntry.RecordId);
        OnAfterPostVend(GenJournalLine, Balancing, TempGLEntryBuf, NextEntryNo, NextTransactionNo);
    end;
```

This request is related to https://github.com/microsoft/ALAppExtensions/issues/24520

**Comment by RadoArvay** [comment]
The new event is already added so I doubt they left 2 events.

**Comment by AndersLarsenMicrosoft** [comment]
My bad. This will be fixed with 24.1 or 24.2 at the latest
